usagov/usagov-2021
# USAGOV-1612 Setup a config split to improve developer workflow and disable devel module on production

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1612

## Description
Require and enable config_split module. Configure it to setup a dev split with: 
1. A complete split on the Devel module.
2. A partial split on the autologout.settings to target the timeout and max_timeout config keys.

Disable this split by default, and explicitly enable the split in settings.local.php.

Additionally, enable the admin_toolbar_search module.
## Type of Changes

- [ X ] New Feature
  - Add screenshot showing what it should look like
- [ X ] Drupal Config (requires "drush cim" TWICE)
- [ X ] New Modules (requires rebuild)

## Testing Instructions


**Note that the configuration must be imported twice for the config_split to fully take effect.**

On local environments:

1. Navigate to /admin/config/people/autologout.
2. The Timeout and Max Timeout values are set to 10000
4. Navigate to /admin/config/development/devel.
5. The Variables Dumper setting has the "symfony var dumper" radio option selected

On dev/stg/prod
1. Navigate to /admin/config/people/autologout.
2. The Timeout and Max Timeout values are set to 1800
3. Navigate to /admin/modules. Local the devel module
4. The box is unchecked, indicating that the module is not installed.
6. Find the new admin toolbar search bar.
7. Use it to locate some items in the admin configuration. 
8. The admin toolbar search works as expected, and there are no php errors on the configuration pages you visit.

### Requires New Config
- [ X ] Yes

### Requires New Content
- [ X ] No

## Security Review


- [ X ] Adds/updates software (including a library or Drupal module)


## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
4. Find the commit of this pull request.
5. Build and deploy the changes.
6. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
